device/ospfv2/interface/interface.yaml

```diff
@@ -0,0 +1,242 @@
+components:
+  schemas:
+    Ospfv2.Interface:
+      description: >-
+        Configuration for single OSPFv2 interface.
+      type: object
+      required: [name,area]
+      properties:
+        name:
+          x-include: ../common/common.yaml#/components/schemas/Named.Object/properties/name
+          x-field-uid: 1
+        choice:
+          description: |-
+            The name of the interface top of which OSPFv2 is configured.
+            - ipv4_name: The globally unique name of the IPv4 interface connected to the DUT.
+              This name must match the "name" field of the "ipv4_addresses" on top which this OSPFv2 interface is configured.
+            - dhcp_intf_name: The Area ID in IPv4 address format.
+          type: string
+          default: ipv4_name
+          x-field-uid: 2
+          x-enum:
+            ipv4_name:
+              x-field-uid: 1
+            dhcp_intf_name:
+              x-field-uid: 2
+        ipv4_name:
+          description: >-
+            The globally unique name of the IPv4 interface connected to the DUT.            
+          type: string
+          x-constraint:
+          - '/components/schemas/Device.Ipv4/properties/name'
+          x-field-uid: 3
+        area:
+          description: >-
+            The OSPF Area ID identifies the routing area to which the host belongs           
+          $ref: '#/components/schemas/Ospfv2Interface.Area'
+          x-field-uid: 5
+        network:
+          description: |- 
+            The OSPF network link type options.
+            - Point to Point: 
+            - Broadcast: 
+            - Point to Multipoint: In this case, at least a neigbor to be configured. 
+          type: string
+          default: broadcast
+          x-field-uid: 6
+          x-enum:
+            broadcast:
+              x-field-uid: 1
+            point_to_point:
+              x-field-uid: 2
+            point_to_multicast:
+              x-field-uid: 3
+        neighbors:
```

Comment:
It is probably better done with a object with default choice and neighbors as attribute.